pkgs/tools/security/knockknock/default.nix

```diff
@@ -1,14 +1,13 @@
 { lib, fetchFromGitHub, python2Packages, hping }:
-let
-  rev  = "bf14bbff";
-in python2Packages.buildPythonApplication rec {
-  pname = "knockknock-r";
-  version = rev;
+
+python2Packages.buildPythonApplication rec {
```

Comment:
Python2 is EOL and there was no update for almost a decade.

If `python3Packages` doesn't work then we should consider to drop `knockknock`.

Reply:
I don't think it is too much of an issue of keeping a package if it is already working though.

If it was something that is broken, I would also prefer to drop it. But it seems to work fine.

Reply:
We are planning to drop python2 at some point and everything that is relying on it should switch rather sooner than later to python3. Also python2Packages.* packages are no longer validated and if something breaks with an upgrade it is acceptable.

Reply:
Yeah, I understand.

Still, until this happens I don't see the issue of allowing this package to build on more platforms. It is not like the diff was huge or many patches were needed.

Actually, most of the changes in this PR were small clean-ups.

Reply:
BTW, I tried to build it with Python 3 before merging this PR. But it obvious just buy looking at the prints as statements (https://github.com/moxie0/knockknock/blob/bf14bbffc5f1d2105cd1d955dabca26b3faa0db4/knockknock.py#L103) that this is never going to work unless someone forks the project and create a Python 3 compatible version.

Reply:
Maybe we can just run the py 2 to 3 conversion script on it and it just works?

Reply:
I don't think this is a good idea since this project doesn't have tests.